### ISSUE TITLE
Server Send Events connections not closing

### ISSUE THREAD

**Issue by bwegman** [SignalR/SignalR#369]
I found two issues with SSE:
1. When using SSE to connect to SignalR from Google Chrome (tested with version 18) upon refreshing the page using SignalR the initial connection to the server remains active until a timeout occurs. This causes issues when using IIS on Win7 as this version is limited to only 10 client connections. Observed behaviour is unresponsiveness from the server until the signalr connections timeout.
2. When using SSE (might also affect other protocols) when a connection is lingering in the situation described above and the application pool is recycled due to updating a binary or touching the web.config the connections remain indefinetely until the w3wp.exe process is exited. This will not happen until IIS decides to recycle the worker process. This seems to be an issue with connections not being force closed server-side upon application recycling. Observed behaviour is multiple connections with very high connection times (300+ seconds) in the Requests overview of IIS (IIS Manager -> Worker Processes -> Requests).

Reproduction scenario for issue 2:
- Setup the QuickStart sample.
- Setup a couple of SignalR connections to your application using the technique described under 1. Make sure SSE is used.
- Either rebuild your webapp or touch the web.config file (trigger a reload of the app domain).
- You might need to repeat the above steps a couple of times.
- One or more connections keep lingering around in the Requests overview.

Both issues do not occur when using longPolling.

A related issue is that connections are not closed correctly upon application recycling when IIS is not immediately closing the app domain. The implementation is incorrectly calling UnregisterObject on IRegisteredObject when the parameter  Immediate for the Stop method is False. It should only do this after finishing clean up. This cleanup however is only triggered when the parameter Immediate is True. For SignalR connections you probably want to close the connections asap as no new connections are accepted by IIS during app domain shutdown. See https://github.com/bwegman/SignalR/commit/ccfcb4a92288b07b5e74362c2e041e931e872d95 for a possible fix for this.

**Comment by ghristov88** [comment]
I think I am having a similar issue with SSE.

1. When I debug on my local machine I have a connection with SSE and then I kill the page, from chrome (48.0.2564.116 m) task manager and then I close the page. The connection does not disconnect at all. I've waited at least 15-20 mins. I checked it in ITransportHeartbeat.GetConnections() and the IsAlive property remains true for the described time. 

I caught this on my live environment where I have 1000+ connections with SSE mostly on mobile devices. The users have multiple connections for the same userID (which should not happen and it doesn't happen for other transports). I guess that on mobile devices they close the tab from the task manager and the same thing happens as if you killed the process (haven't been able to reproduce it).

2. There is another thing that our QA team simulated, not sure if is related. When a mobile device is connected with SSE, then the browser is minimized, then the device is locked (iPhone) or the internet is stopped (Android tablet) and after 30 seconds it is brought back, we end up with a new connection on SSE and the old one is still alive and keeps being alive after 10-15 mins. I haven't had the time to simulate it in an isolated project, but I decided to write it anyway, because it might help knowing about it regarding the first issue. 

We are using signalR  2.2.0 server and js client.
